### PR TITLE
Add CI config for github actions

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Currently, we are not running any steps in CI for UTO

closes: #1724 

## What is the new behavior?
We added the GitHub actions config needed for CI to run Tests 

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: .github/workflows/ci-uto.yaml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  push:
+    branches: [ 'main', 'release-**' ]
+    paths:
+      - 'python-sdk/**'
+      - 'universal_transfer_operator/**'
+
+  pull_request:
+    branches: [ 'main', 'release-**' ]
+    paths:
+      - 'python-sdk/**'
```
Comment (tatiana) [inline .github/workflows/ci-uto.yaml:12]: Why do we want to trigger the UTO tests when there is a change to the `python-sdk`?

File: .github/workflows/ci-uto.yaml [hunk 2]
```diff
@@ -0,0 +1,189 @@
+name: Build and test Universal Transfer Operator
+on:
+  push:
+    branches: [ 'main', 'release-**' ]
+    paths:
+      - 'python-sdk/**'
```
Comment (tatiana) [inline .github/workflows/ci-uto.yaml:6]: Why do we want to trigger the UTO tests when there is a change to the python-sdk?

**Review by tatiana — COMMENTED** [review]
WDYT of labelling UTO PRs in a similar way to what we do with SQL CLI ones?
https://github.com/astronomer/astro-sdk/blob/main/.github/labeler.yml

File: .github/workflows/ci-uto.yaml (hunk 2 above)
Comment (utkarsharma2) [reply .github/workflows/ci-uto.yaml:6]: Hmm...yes should remove it. I copied this file from SQL-CLI. I guess we can remove it from there as well right? 

https://github.com/astronomer/astro-sdk/blob/361031295b4eec77cd6de0607b8648968a8c3c6c/.github/workflows/ci-sql-cli.yaml#L6

Comment (utkarsharma2) [inline .github/workflows/ci-uto.yaml:6]:
```suggestion
```

File: .github/workflows/ci-uto.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+on:
+  push:
+    branches: [ 'main', 'release-**' ]
+    paths:
+      - 'universal_transfer_operator/**'
+
+  pull_request:
+    branches: [ 'main', 'release-**' ]
+    paths:
+      - 'python-sdk/**'
```
Comment (utkarsharma2) [inline .github/workflows/ci-uto.yaml:11]:
```suggestion
```

**Review by tatiana — APPROVED** [review]
@utkarsharma2, as we discussed, I'm happy for us to have separate tickets/PRs for fixing the following checks:
- Fix type check
- Code coverage

I'm happy for the PR to be merged once Deep Source and the other checks pass.

**Comment by utkarsharma2** [comment]
@tatiana Added tickets  
1. https://github.com/astronomer/astro-sdk/issues/1788 
2. https://github.com/astronomer/astro-sdk/issues/1787

**Pull request merged**